Services/Navigation/classes/class.ilNavigationHistoryGUI.php

```diff
@@ -152,13 +158,15 @@ function handleNavigationRequest()
 	 * @param
 	 * @return
 	 */
-	function removeEntries()
+	public function removeEntries()
 	{
 		$ilNavigationHistory = $this->nav_history;
-		
+
 		$ilNavigationHistory->deleteDBEntries();
 		$ilNavigationHistory->deleteSessionEntries();
+
+		$script = $this->http->request()->getHeader('Referer')[0];
```

Comment:
@alex40724 the magic is here, after removing the entries I just redirect to the last location.

Reply:
Won't this fail on POST requests (commands not part of the Referer)?

Reply:
valid point... there are a lot of those situations in ILIAS. did not consider that. I‘ll look for another solution then